tests/Vimeo/VimeoTest.php

```diff
@@ -0,0 +1,253 @@
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoUploadException
+     */
+    public function testUploadWithNonExistedFile()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->upload('./the_file_is_invalid');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoUploadException
+     */
+    public function testUploadWithFakeMachineIdShouldReturnVimeoRequestException()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->upload(__DIR__.'/../../composer.json', 'fake_machine_id');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoUploadException
+     */
+    public function testReplaceWithNonExistedFile()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->replace('https://vimeo.com/241711006', './the_file_is_invalid');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoRequestException
+     */
+    public function testReplaceWithVideoUriAndMachineIdShouldReturnVimeoRequestException()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->replace('https://vimeo.com/241711006', __DIR__.'/../../composer.json', 'fake_machine_id');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoUploadException
+     */
+    public function testUploadImageWithNonExistedFile()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->uploadImage('https://vimeo.com/241711006', './the_file_is_invalid');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoRequestException
+     */
+    public function testUploadImageWithPictureUriAndMachineIdShouldReturnVimeoRequestException()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->uploadImage('https://vimeo.com/user59081751', __DIR__.'/../../composer.json', 'fake_machine_id');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoUploadException
+     */
+    public function testuploadTexttrackWithNonExistedFile()
+    {
+        // Arrange
+        $vimeo = new Vimeo($this->clientId, $this->clientSecret);
+
+        // Act
+        $result = $vimeo->uploadTexttrack('https://vimeo.com/241711006', './the_file_is_invalid', 'fake_track_type', 'zh_TW');
+    }
+
+    /**
+     * @expectedException Vimeo\Exceptions\VimeoRequestException
+     */
+    public function testuploadTexttrackWithPictureUriAndMachineIdShouldReturnVimeoRequestException()
```

Comment:
Uppercase `upload` here so the test is `testUploadTexttrackWithPictureUriAndMachineIdShouldReturnVimeoRequestException`.